### PR TITLE
Add TypeScript Type Definition File

### PR DESCRIPTION
**Pull request by kinyat**
Say hello to TypeScript :)

### CODE REVIEW

File: src/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+declare module 'type-to-reducer' {
+  export default function typeToReducer (
+    reducerMap: any,
```
Comment (tomatau) [inline src/index.d.ts]: Maybe reducerMap should be an object of reducerMaps or functions rather than an any?

File: src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+    initialState: any
+  ): any
```
Comment (tomatau) [inline src/index.d.ts]: it always returns a function

**Comment by tomatau** [comment]
Cool thanks! I've left 2 comments but happy to merge if you think they're not important

**Comment by kinyat** [comment]
@tomatau Updated. This should be good enough to make it works in TypeScript for some value type checking (the value can be either an array or a function with a state and action)

**Comment by tomatau** [comment]
Nice! Ty

**Comment by tomatau** [comment]
published as 1.0.3

**Pull request merged**